packages/react/menu/src/Menu.tsx

```diff
@@ -553,9 +608,12 @@ const ITEM_NAME = 'MenuItem';
 const ITEM_DEFAULT_TAG = 'div';
 const ITEM_SELECT = 'menu.itemSelect';
 
-type MenuItemOwnProps = Polymorphic.Merge<
-  Polymorphic.OwnProps<typeof MenuItemImpl>,
-  { onSelect?: (event: Event) => void }
+type MenuItemOwnProps = Omit<
+  Polymorphic.Merge<
+    Polymorphic.OwnProps<typeof MenuItemImpl>,
+    { onSelect?: (event: Event) => void }
+  >,
+  keyof MenuItemImplPrivateProps
 >;
```

Comment:
```suggestion
type MenuItemOwnProps = Polymorphic.Merge<
  Omit<Polymorphic.OwnProps<typeof MenuItemImpl>, keyof MenuItemImplPrivateProps>,
  { onSelect?: (event: Event) => void }
>;
```

Just means we're only omitting from the bit where they exist. Reduces some of the wrapping too hahaha, and you know how i feel about wrapping 😅 

Although, we probably don't need this anymore if we do the `onKeyCapture` thing.

Reply:
Haha, wrapping or not I think that just reads better! I'll apply it for the time being.